src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaFactory.java

```diff
@@ -29,4 +31,8 @@ public NakadiKafkaConsumer createNakadiConsumer(final String topic, final List<K
         return new NakadiKafkaConsumer(getConsumer(), topic, kafkaCursors, pollTimeout);
     }
 
+    @Override
+    public void updateProperties(final Properties kafkaProperties) {
+        kafkaProducer = new KafkaProducer<>(kafkaProperties);
```

Comment:
a few issues that might or might not relevant:
1. what i can't follow from this is: how this actually changes the kafkaProducer in the repository? there createProducer gave a reference to the previous kafkaProducer and there it was not changed by this update until afurther createConsumer, which does not happen.
2. allright that @Scheduled is thread safe, but kafkaProducer here is not really safe for access, right? the updateProperties too can theoretically be called by other means that will mess things up. wouldn't it make sense to lock updateProperties to write wrt to reads on createProducer?
3. does it make sense to offer the possibility to explicitly trigger this reloading? this might come in handy...